check_source.py

```diff
@@ -152,10 +152,15 @@ def check_source_submission(self, source_project, source_package, source_revisio
             if devel_project:
                 if ((source_project != devel_project or source_package != devel_package) and
                    not (source_project == target_project and source_package == target_package)):
-                    # Not from proper devel project/package and not self-submission.
-                    self.review_messages['declined'] = 'Expected submission from devel package %s/%s' % (
-                        devel_project, devel_package)
-                    return False
+                    # check if the devel project & package are using scmsync
+                    # => waive the devel project source submission requirement
+                    meta = ET.fromstringlist(show_package_meta(self.apiurl, devel_project, devel_package))
+                    scm_sync = meta.find('scmsync')
+                    if not scm_sync or (scm_sync and not scm_sync.text.startswith(f"https://src.opensuse.org/pool/{source_package}")):
+                        # Not from proper devel project/package and not self-submission.
+                        self.review_messages['declined'] = 'Expected submission from devel package %s/%s' % (
```

Comment:
maybe a check on the user being the scm-sync bot user? so that we don't allow submitrequests from *not* the bot user accidentally (that bypass the git workflow)

Reply:
That will require quite a bit of refactoring, as the information who submitted the request is not forwarded into the individual `check_action_*` functions.

Reply:
That sucks

Reply:
can we check that the submitting source project is coming from home:defolos or whatever the trusted prefix will be? maybe we use a reasonable prefix that isn't home: and that we can actually keep..

Reply:
Turns out it wasn't that much refactoring work after all, I've implemented checking for the user now as well.

Reply:
e.g. check for src_project == `devel:Factory:git-workflow:staging` and then we use that one for submission source?